x-pack/plugins/observability_ai_assistant/public/components/insight/insight.tsx

```diff
@@ -142,18 +213,87 @@ export function Insight({ messages, title, dataTestSubj }: InsightProps) {
     [service]
   );
 
+  const handleSend = (newPrompt: string) => {
+    const clonedMessages = cloneDeep(messages);
```

Comment:
In all usages of the insights so far it is the first and only message. I went the clonedeep route in this case since it wouldn't be more costly on such a small dataset and was straightforward to implement

Reply:
Merging, happy to correct this in a follow up